### PR TITLE
Fix correct usage of angular velocity in update_odometry() function

### PR DESCRIPTION
**Pull request by MCFurry**
Thanks for this nice steering_controllers_library!

Using it however, we noticed issues in the odometry calculation where our steered robot would spin incredibly fast.
There seems a small math error in the calculation which this PR fixes:
The integration expects distances and angles, while the accumulators expect a velocity.

Probably also relates to https://github.com/ros-controls/ros2_controllers/issues/937

### CODE REVIEW

**Review by saikishor — COMMENTED** [review]
I haven't reviewed the original code yet, but isn't the angle here referring to the angle itself instead of angular velocity?

**Comment by MCFurry** [comment]
Well in all the functions that call `update_odometry(linear_velocity, angular, dt)`, the angular input is calculated as:
`tan(steer_pos_) * linear_velocity / wheelbase_` which has a velocity as outcome.
So it seems that angular is indeed a velocity here.

**Comment by saikishor** [comment]
> Well in all the functions that call `update_odometry(linear_velocity, angular, dt)`, the angular input is calculated as:
> `tan(steer_pos_) * linear_velocity / wheelbase_` which has a velocity as outcome.
> So it seems that angular is indeed a velocity here.

Fine! Thanks for the explanation. Do you mind adding tests for your changes?

**Comment by MCFurry** [comment]
> > Well in all the functions that call `update_odometry(linear_velocity, angular, dt)`, the angular input is calculated as:
> > `tan(steer_pos_) * linear_velocity / wheelbase_` which has a velocity as outcome.
> > So it seems that angular is indeed a velocity here.
> 
> Fine! Thanks for the explanation. Do you mind adding tests for your changes?

Sure thing! I added some very simple tests, please let me know if these are okay or if we want to test some more exotic situations?

**Comment by christophfroehlich** [comment]
Please have a look at https://github.com/ros-controls/ros2_controllers/pull/954 and review what I've written there. I realized some wrong code too, but wanted to have the theory/nomenclature approved before I start fixing/refactoring.

**Comment by MCFurry** [comment]
> Please have a look at #954 and review what I've written there. I realized some wrong code too, but wanted to have the theory/nomenclature approved before I start fixing/refactoring.

Very nice! I'll review that soon!

**Comment by MCFurry** [comment]
@christophfroehlich I just reviewed your documentation effort and that looks really nice! Great work!
Do you think merging this PR in the meantime would be a small step forwards already? It might make the Jazzy release still?

**Comment by christophfroehlich** [comment]
I'd like to update the function documentation/docstring to make this point clear. But you deactivated the tick  _Maintainers are allowed to edit this pull request._ I'll open a new PR on top of that one

**Comment by MCFurry** [comment]
> I'd like to update the function documentation/docstring to make this point clear. But you deactivated the tick _Maintainers are allowed to edit this pull request._ I'll open a new PR on top of that one

Ahh I wasn't aware of that checkbox, it seems I also can't change that anymore?

**Comment by Timple** [comment]
@christophfroehlich Can you list the changes you wanted to make? Or create a pull request targetting this PR?
That way this PR can keep moving forward :slightly_smiling_face:

**Review by christophfroehlich — APPROVED** [review]
Thanks for the changes, the algorithm seems to be fine now. I made a follow-up PR #1149 to get a more consistent nomenclature.

**Comment by christophfroehlich** [comment]
@mergifyio backport humble iron

File: steering_controllers_library/test/test_steering_odometry.cpp
Comment (christophfroehlich) [inline steering_controllers_library/test/test_steering_odometry.cpp:142]:
```suggestion
  EXPECT_NEAR(odom.get_linear(), 1.0, 1e-3);
```
satisfying the linters.

**Comment by christophfroehlich** [comment]
@Timple @MCFurry please have a look at the failing pre-commit jobs (a missing newline at the end of a file). The other failing jobs are not related. fyi, we need another review of someone with write access (@bmagyar or @destogl). In the meantime, could you maybe leave a review on #954? ;)

File: steering_controllers_library/src/steering_odometry.cpp
Comment (saikishor) [inline steering_controllers_library/src/steering_odometry.cpp:52]:
```suggestion
  const double linear_velocity, const double angular_velocity, const double dt)
```

Shall we take this as an opportunity to rename the function variable to angular_velocity to maintain consistency?

@christophfroehlich what do you think?

**Review by saikishor — APPROVED** [review]
The current fix and the tests look good to me.

Thank you 👍🏽

File: steering_controllers_library/src/steering_odometry.cpp
Comment (christophfroehlich) [reply steering_controllers_library/src/steering_odometry.cpp:52]: https://github.com/ros-controls/ros2_controllers/pull/1149

Comment (saikishor) [reply steering_controllers_library/src/steering_odometry.cpp:52]: Awesome. Thank you

**Comment by MCFurry** [comment]
> I'd like to update the function documentation/docstring to make this point clear. But you deactivated the tick _Maintainers are allowed to edit this pull request._ I'll open a new PR on top of that one

Ahh I wasn't aware of that checkbox, it seems I also can't change that anymore?

> @Timple @MCFurry please have a look at the failing pre-commit jobs (a missing newline at the end of a file). The other failing jobs are not related. fyi, we need another review of someone with write access (@bmagyar or @destogl). In the meantime, could you maybe leave a review on #954? ;)

Done!

**Review by bmagyar — APPROVED** [review]

**Pull request merged**
